### PR TITLE
add work-around for systems without fseeko() (such as Android)

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

File: include/libtorrent/config.hpp [hunk 1]
```diff
@@ -175,6 +175,10 @@ POSSIBILITY OF SUCH DAMAGE.
 #if defined __ANDROID__
 #define TORRENT_ANDROID
 #define TORRENT_HAS_FALLOCATE 0
+#if __ANDROID_API__ < 21
```
Comment (aldenml) [inline include/libtorrent/config.hpp:178]: sorry I didn't mention the right api level at first. This needs to be `24`.

**Review by aldenml — COMMENTED** [review]
looks good

**Comment by arvidn** [comment]
@aldenml is there an easy-to-install android cross-compiler I could run in CI?

**Comment by aldenml** [comment]
not that I'm aware of, but it's easy to put together another GitHub action similar to what I have. Problem is that I don't think it's possible to run executables. Are you open to just compile and perform no test for android?

**Comment by arvidn** [comment]
I think just compiling is better than nothing

**Comment by aldenml** [comment]
good, I will help with that. If you want to support lower APIs (< 24), you also need hacks around ifaddrs and random. Do you have an opinion already?

**Comment by arvidn** [comment]
> good, I will help with that. If you want to support lower APIs (< 24), you also need hacks around ifaddrs and random. Do you have an opinion already?

I don't have a strong opinion. Do you have patches already?

**Comment by aldenml** [comment]
I will try to send a PR to build with no hacks (API 28), then start lowering with workarounds :)

**Pull request merged**